src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts

```diff
@@ -17,11 +18,27 @@ provideEditingConfig(DataLabelWidget, {
         { value: 'link', title: 'Link' },
       ],
     },
+    datetimeFormat: {
+      title: 'Date format',
+      description: 'Default: Relative',
+      values: [
+        { value: 'relative', title: 'Relative' },
+        { value: 'datetime', title: 'Date and time' },
+        { value: 'date', title: 'Only date' },
+      ],
+    },
     marginBottom: {
       title: 'Add margin bottom?',
     },
   },
-  properties: ['attributeName', 'showAs', 'valueSize', 'marginBottom'],
+  properties: (widget) =>
+    [
+      'attributeName',
+      'showAs',
+      widget.get('showAs') === 'datetime' ? 'datetimeFormat' : null,
+      'valueSize',
+      'marginBottom',
+    ].filter(isString),
```

Comment:
* [ ] FYI I'm skeptical about adding dependencies for trivial stuff like `].filter(p => p !== null),`

Reply:
I already tried several things, e.g.

```ts
  properties: (widget): string[] =>
    [
      'attributeName',
      'showAs',
      widget.get('showAs') === 'datetime' ? 'datetimeFormat' : null,
      'valueSize',
      'marginBottom',
    ].filter((p) => p !== null),
```

or
```ts
  properties: (widget): string[] =>
    [
      'attributeName',
      'showAs',
      widget.get('showAs') === 'datetime' ? 'datetimeFormat' : null,
      'valueSize',
      'marginBottom',
    ].filter((p) => typeof p === 'string'),
```

All of these still print:

```
Type '(string | null)[]' is not assignable to type 'string[]'.
Type 'string | null' is not assignable to type 'string'.
Type 'null' is not assignable to type 'string'.
```

With `isString` (or probably some other guard function) it works.

Reply:
I see, TS stuff.
The cleanest I can think of right now is an inline type guard:
```ts
].filter((p): p is string => typeof p === 'string'),
```

Reply:
=> https://github.com/Scrivito/scrivito-portal-app/pull/333